.github/actions/test-dotnet/Invoke-SolutionTests.ps1

```diff
@@ -43,27 +43,21 @@ foreach ($test in $tests) {
         '--no-build',
         '--nologo',
         '--logger', 'trx;LogFileName=test-results.trx'
+        # This is for xUnit ITestOutputHelper, see https://xunit.net/docs/capturing-output.
+        '--logger', 'console;verbosity=detailed'
         '--verbosity', $Verbosity
         $Filter ? '--filter' : ''
         $Filter ? $Filter : ''
         $test
     )
 
-    dotnet test @dotnetTestSwitches 2>&1 >test.out
+    dotnet test @dotnetTestSwitches
 
     if ($?)
     {
         Write-Output "Test Successful: $test"
         continue
     }
 
-    $needsGrouping = (Select-String "::group::" test.out).Length -eq 0
-
-    if ($needsGrouping) { Write-Output "::group::Test Failed: $test" }
-
-    bash -c "cat test.out | grep -v 'Connection refused \[::ffff:127.0.0.1\]' | grep -v 'ChromeDriver was started successfully'"
```

Comment:
Could we still filter out these pointless "Connection refused" and "ChromeDriver" log lines on `dotnet test @dotnetTestSwitches`?

Reply:
I don't know or have found a way to live-filter a streaming output, though I only have smallish experience with PS.

The goal here is for the output of the tests to show up in the log continuously (to pinpoint what the last thing was before the runner was killed due to timeout). If there's a file at the end of the `dotnet test` call then you can grep like this, but we can't wait for a file.

We could still redirect to a file and tail it and that [can be filtered](https://stackoverflow.com/a/48773243/220230) but going through a file doesn't seem like the best idea, and we also need to somehow know when the stream actually ended and quit the `Wait`.

So, I don't know. It seems to me that implementing grouping around the test project if it doesn't have its custom group and letting UI tests group themselves, as well as filtering out such messages need a file. If we have a file, however, we won't have a continuous output.

Do you have any suggestions by chance?

BTW I'm also trying to figure out where the annoying log messages come from and disable them in the first place.

Reply:
`dotnet test @dotnetTestSwitches 2>&1 | ? { $_ -notlike '*Connection refused [[]::ffff:127.0.0.1[]]*' -and $_ -notlike '*ChromeDriver was started successfully*' }` should filter it line-by-line as it's passed to the pipe. We can live without the outer grouping until they finally add named groups to the runner.

Reply:
Ah nice, so this works like this!